src/include/buffer.h

```diff
@@ -50,10 +50,13 @@
 #include <exception>
 #include <type_traits>
 
+#include <boost/system/system_error.hpp>
```

Comment:
this at least add a build-time dependency on libboost-{dev,devel} to librados-{dev,devel}. and it breaks the test of http://pulpito.ceph.com/kchai-2020-01-14_13:07:57-rados-wip-kefu-testing-2020-01-14-1719-distro-basic-smithi/4667375/

Reply:
why does it break this test?

Reply:
for that matter, why do we object to a build-time dependency on libboost-dev?

Reply:
@mattbenjamin

> why/how does it break this test?
```
2020-01-14T14:16:54.200 INFO:tasks.workunit.client.0.smithi132.stderr:In file included from /usr/include/rados/librados.hpp:11,
2020-01-14T14:16:54.200 INFO:tasks.workunit.client.0.smithi132.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/include/rocksdb/utilities/env_librados.h:14,
2020-01-14T14:16:54.200 INFO:tasks.workunit.client.0.smithi132.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/utilities/env_librados.cc:4:
2020-01-14T14:16:54.200 INFO:tasks.workunit.client.0.smithi132.stderr:/usr/include/rados/buffer.h:53:10: fatal error: boost/system/system_error.hpp: No such file or directory
2020-01-14T14:16:54.200 INFO:tasks.workunit.client.0.smithi132.stderr: #include <boost/system/system_error.hpp>
2020-01-14T14:16:54.200 INFO:tasks.workunit.client.0.smithi132.stderr:          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2020-01-14T14:16:54.201 INFO:tasks.workunit.client.0.smithi132.stderr:compilation terminated.
2020-01-14T14:16:54.201 INFO:tasks.workunit.client.0.smithi132.stderr:make[3]: *** [CMakeFiles/rocksdb-shared.dir/build.make:3209: CMakeFiles/rocksdb-shared.dir/utilities/env_librados.cc.o] Error 1
2020-01-14T14:16:54.202 INFO:tasks.workunit.client.0.smithi132.stderr:make[3]: *** Waiting for unfinished jobs....
2020-01-14T14:16:54.437 INFO:tasks.workunit.client.0.smithi132.stderr:make[2]: *** [CMakeFiles/Makefile2:2145: CMakeFiles/rocksdb-shared.dir/all] Error 2
2020-01-14T14:16:54.437 INFO:tasks.workunit.client.0.smithi132.stderr:make[1]: *** [CMakeFiles/Makefile2:6356: CMakeFiles/rocksdb_env_librados_test.dir/rule] Error 2
2020-01-14T14:16:54.438 INFO:tasks.workunit.client.0.smithi132.stderr:make: *** [Makefile:2177: rocksdb_env_librados_test] Error 2
```

> for that matter, why do we object to a build-time dependency on libboost-dev?

 i am not for or against adding a build-time dependency on libboost-dev by post this comment. i was just trying to remind Adam that we need to at least add libboost-dev to the dependency list of librados-dev. otherwise the test fails.

but a small set of dependency is always a good thing.

Reply:
i look forward to std::net eventually replacing our use of boost::asio so we can drop all of this boost::error_code stuff

Reply:
@tchaikov Thanks! I'm going to try and move this include dependency out of here and into common/error_code

Reply:
@cbodley Boost finally added is_fail which I've wanted for RGW, though I'm not sure if I want it enough to keep using something other than std::error_code.